module.go

```diff
@@ -86,65 +123,80 @@ func (m *Module) Run(ctx context.Context) error {
 		return xerrors.Errorf("setting up pubsub: %w", err)
 	}
 
-	sub, err := m.topic.Subscribe()
+	sub, err := m.client.topic.Subscribe()
 	if err != nil {
 		return xerrors.Errorf("subscribing to topic: %w", err)
 	}
 
+	messageQueue := make(chan *gpbft.GMessage, 20)
+	m.client.messageQueue = messageQueue
+
+	h, err := newHost(m.id, m.Manifest, m.client)
+	if err != nil {
+		return xerrors.Errorf("creating gpbft host: %w", err)
+	}
+
+	go func() {
+		err := h.Run(ctx)
+		m.log.Errorf("running host: %+v", err)
```

Comment:
We should:

1. Cancel `ctx` from the outer module on exit (e.g., on error).
2. Send the error on some channel if there's a failure (or close the channel to indicate that we've exited.